tests/test_class_sh_shared_ptr_copy_move.cpp

```diff
@@ -0,0 +1,71 @@
+#include "pybind11_tests.h"
+
+#include <pybind11/smart_holder.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace pybind11_tests {
+namespace {
+
+const std::string fooNames[] = {"ShPtr_", "SmHld_"};
+
+template <int SerNo>
+struct Foo {
+    std::string hisotry;
+    Foo(const std::string &hisotry_) : hisotry(hisotry_) {}
+    Foo(const Foo &other) = delete;
+    Foo(Foo &&other)      = delete;
+    std::string get_history() const { return "Foo" + fooNames[SerNo] + hisotry; }
+};
+
+using FooShPtr = Foo<0>;
+using FooSmHld = Foo<1>;
+
+} // namespace
+} // namespace pybind11_tests
+
+PYBIND11_TYPE_CASTER_BASE_HOLDER(pybind11_tests::FooShPtr)
+PYBIND11_SMART_HOLDER_TYPE_CASTERS(pybind11_tests::FooSmHld)
```

Comment:
I overlooked this before, this needs to be (untested but I hope it's correct; there are many examples in e.g. test_smart_ptr.cpp):
```
PYBIND11_TYPE_CASTER_BASE_HOLDER(pybind11_tests::FooShPtr, std::shared_ptr<pybind11_tests::FooShPtr>)
```

Reply:
Fixed.